sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { toBase64 } from '@mysten/bcs';
+import { secp256r1 } from '@noble/curves/p256';
+import { blake2b } from '@noble/hashes/blake2b';
+import { randomBytes } from '@noble/hashes/utils';
+import type {
+	AuthenticationCredential,
+	RegistrationCredential,
+} from '@simplewebauthn/typescript-types';
+
+import { PasskeyAuthenticator } from '../../bcs/bcs.js';
+import type { IntentScope, SignatureWithBytes } from '../../cryptography/index.js';
+import { messageWithIntent, SIGNATURE_SCHEME_TO_FLAG, Signer } from '../../cryptography/index.js';
+import type { PublicKey } from '../../cryptography/publickey.js';
+import type { SignatureScheme } from '../../cryptography/signature-scheme.js';
+import {
+	parseDerSPKI,
+	PASSKEY_PUBLIC_KEY_SIZE,
+	PASSKEY_SIGNATURE_SIZE,
+	PasskeyPublicKey,
+} from './publickey.js';
+
+export interface PasskeyCreateOptions {
+	timeout: number; // The timeout for the passkey creation dialog.
+	name: string; // The name of the user
+	displayName: string; // The display name of the user, this is shown at creation dialog
+	rpName: string; // The name of the relying party, this is shown to user in the passkey creation dialog.
```

Comment:
I think these are going away in jordans refactor, but for for if you add these as doc comments, they will show up in editors:

eg:

```ts
/** The name of the relying party, this is shown to user in the passkey creation dialog. */
rpName: string; 
```